src/cryptography/x509.py

```diff
@@ -1318,6 +1322,11 @@ def __init__(self, key_identifier, authority_cert_issuer,
         self._authority_cert_issuer = authority_cert_issuer
         self._authority_cert_serial_number = authority_cert_serial_number
 
+    @classmethod
+    def from_issuer_certificate(cls, certificate):
+        digest = _key_identifier_from_public_key(certificate.public_key())
+        return cls(digest, None, None)
```

Comment:
Most implementations don't populate this, but it is possible to do so. I have no objection to changing it to from_public_key (and we can add a from_issuer_certificate later if people ask for it)